conda_smithy/utils.py

```diff
@@ -20,6 +21,19 @@
 CONDA_BUILD = "conda-build"
 
 
+class RecipeVersion(IntEnum):
+    V1 = 1
+    V2 = 2
+
+    @property
+    def is_v1(self):
+        return self == RecipeVersion.V1
+
+    @property
+    def is_v2(self):
+        return self == RecipeVersion.V2
```

Comment:
I don't see the value in having these when one can simply do `recipe_version == 1`. Are expecting to have minor versions too? Like `2.1` or `(2, 1)`?  I'm concerned about having to add (and check for) `is_v3` in the future, and so on.

Reply:
I am unsure if you are arguing against the functions (`is_v*`) or against the enum as such?

I think an enum is better for typing consistency. I don't think the functions hurt.

Reply:
I think 
`if recipe_version.is_v1:`
express better domain when reading code rather than:
if `recipe_version == 1`. 

Also, Enum give us type soundness for further iterations and will allow us to extend it and add v3

Reply:
Yeah I don't get the enum thing here at all. A python int is perfectly acceptable and simpler for new devs to understand.

Reply:
Coming from Rust, it's very normal to even create "wrapper" types over strings or things like that to give the type a shape.

Enums are nice, because type checkers can make sure that they were used exhaustively. For example, if we would be able to use the `match` statement, we could write something like:

```
match recipe_version:
  V1: do_foo()
  V2: do_baz()
```

And get a warning as soon as we are starting to add support for `V3`.

Reply:
Maybe some other reasons: 
An integer is just a mathematical thing. We want something "categorical". Your not supposed to do mathematical operations on the int. It's fundamentally the wrong "type".

Readability: Enums provide meaningful names for constant values, making code more self-documenting and easier to understand.
Type safety: Enums create a distinct type, preventing accidental comparisons with unrelated integers.
Namespace: Enums group related constants together, reducing global namespace pollution.
IDE support: Most IDEs offer better autocomplete and error checking for enums.
Iteration: Enums can be easily iterated over, listing all possible values.
Serialization: Enums often have better serialization support in various libraries.
Maintenance: Changing an enum value updates all references, unlike hardcoded integers.

Reply:
> Yeah I don't get the enum thing here at all. A python int is perfectly acceptable and simpler for new devs to understand.

I would like to add that it's very hard to represent with `: int ` type all versions that we want and will support.

So reading:
`recipe_version: int = 2`, it's hard to get what are the all `recipe_versions` that we support, and later it will make refactoring harder to do

Reply:
Given how much work moving from one version to the next is going to be, I am not in favor of ever doing it again. If we really think we will need to do it again, then we need to modify the v2 spec to fix w/e we want fixed now. 

Given this, I expect an int will serve us just fine.

Reply:
But then a bool would also have worked :D

Reply:
Yep. I'm for either the book or an int. The whole enum thing is forcing patterns from rust into python imho which doesn't make sense.

Reply:
@beckermr @wolfv  I've changed from Enum to int